metadata-ingestion/src/datahub/ingestion/source/ge_profiling_config.py

```diff
@@ -136,41 +142,46 @@ def ensure_field_level_settings_are_normalized(
-        table_level_profiling_only_key = "profile_table_level_only"
         max_num_fields_to_profile = values.get(max_num_fields_to_profile_key)
-        if values.get(table_level_profiling_only_key):
-            all_field_level_metrics: List[str] = [
-                "include_field_null_count",
-                "include_field_distinct_count",
-                "include_field_min_value",
-                "include_field_max_value",
-                "include_field_mean_value",
-                "include_field_median_value",
-                "include_field_stddev_value",
+
+        # Disable all field-level metrics.
+        if values.get("profile_table_level_only"):
+            for field_level_metric in cls.__fields__:
+                if field_level_metric.startswith("include_field_"):
+                    values.setdefault(field_level_metric, False)
+
+            assert (
+                max_num_fields_to_profile is None
+            ), f"{max_num_fields_to_profile_key} should be set to None"
+
+        # Disable expensive queries.
+        if values.get("turn_off_expensive_profiling_metrics"):
+            expensive_field_level_metrics: List[str] = [
                 "include_field_quantiles",
                 "include_field_distinct_value_frequencies",
                 "include_field_histogram",
                 "include_field_sample_values",
             ]
-            # Suppress all field-level metrics
-            for field_level_metric in all_field_level_metrics:
-                values[field_level_metric] = False
-            assert (
-                max_num_fields_to_profile is None
-            ), f"{max_num_fields_to_profile_key} should be set to None"
+            for expensive_field_metric in expensive_field_level_metrics:
+                values.setdefault(expensive_field_metric, False)
 
-        if values.get("turn_off_expensive_profiling_metrics"):
-            if not values.get(table_level_profiling_only_key):
-                expensive_field_level_metrics: List[str] = [
-                    "include_field_quantiles",
-                    "include_field_distinct_value_frequencies",
-                    "include_field_histogram",
-                    "include_field_sample_values",
-                ]
-                for expensive_field_metric in expensive_field_level_metrics:
-                    values[expensive_field_metric] = False
-            if max_num_fields_to_profile is None:
-                # We currently profile up to 10 non-filtered columns in this mode by default.
-                values[max_num_fields_to_profile_key] = 10
+            # By default, we profile at most 10 non-filtered columns in this mode.
+            values.setdefault(max_num_fields_to_profile_key, 10)
 
         return values
+
+    def any_field_level_metrics_enabled(self) -> bool:
+        return any(
+            getattr(self, field_name)
+            for field_name in self.__fields__
+            if field_name.startswith("include_field_")
```

Comment:
Maybe later on, it would make sense to have these as constants as it is less prone to typos and easier to refactor if needed.